postfix/files/main.cf

```diff
@@ -1,17 +1,7 @@
 {%- from "postfix/map.jinja" import postfix with context -%}
 {%- set config = salt['pillar.get']('postfix:config', {}) -%}
 
-{%- if not salt['pillar.get']('postfix:mapping', False) %}
-{#- Let the user configure mapping manually. -#}
-{%- set processed_parameters = [] %}
-{%- else -%}
-{#- TODO: alias_maps probably belongs here, too: #}
-{%- set processed_parameters = [
-        'virtual_alias_maps',
-        'smtp_sasl_password_maps',
-        'sender_canonical_maps',
-    ] %}
-{%- endif -%}
+{%- set processed_parameters = salt['pillar.keys']('postfix:mapping') | list %}
```

Comment:
@alxwr Would you mind to add a comment stating why we need `list` filter here?
We know, because of Python 3.6, but others would appreciate that greatly. :smiley:

Reply:
> @alxwr Would you mind to add a comment stating why we need `list` filter here?
> We know, because of Python 3.6, but others would appreciate that greatly. smiley

Added comment. :-)

Reply:
Looks like you also need to prepend the `|default([], true)` filter before the `list`, since if the Pillar does not exist, the renderer returns `KeyError: u'Pillar key not found: postfix:mapping'`.

Reply:
@vutny good catch! :-) Updated the PR.

Reply:
@vutny We still get that error even with `|default([], true)`:

`KeyError: 'Pillar key not found: postfix:mapping'`

I'd switch back to

```sls
{%- set processed_parameters = salt['pillar.get']('postfix:mapping', {}).keys() | list %}
```

Reply:
Looks like having `pillar.keys` here is not an option here. The exception is raised on the Salt module level, so the `default` filter is no use. I guess we need to revert back to the `get` method and either continue with dict to avoid `dict_keys object` error, or convert it to a list.

Reply:
Updated the PR.

Reply:
I'd stick with the list. After all its usable in Py2.7 and Py3.x without jumping through loops.